apps/web/src/components/tx/security/shared/TxSecurityContext.tsx

```diff
@@ -56,11 +58,26 @@ export const TxSecurityContext = createContext<TxSecurityContextProps>(defaultSe
 
 export const TxSecurityProvider = ({ children }: { children: ReactElement }) => {
   const { safeTx, safeMessage, txOrigin } = useContext(SafeTxContext)
-  const [blockaidResponse, blockaidError, blockaidLoading] = useBlockaid(safeTx ?? safeMessage, txOrigin)
+  const txData = useDebounce(safeTx ?? safeMessage, 300)
+  const [blockaidResponse, blockaidError, blockaidLoading] = useBlockaid(txData, txOrigin)
+  const reportScan = useBlockaidReportScan(blockaidResponse?.payload?.requestId)
+  const needsRiskConfirmation = !!blockaidResponse && blockaidResponse.severity >= SecuritySeverity.HIGH
 
   const [isRiskConfirmed, setIsRiskConfirmed] = useState(false)
   const [isRiskIgnored, setIsRiskIgnored] = useState(false)
 
+  // Report scan status when risk is confirmed or ignored
+  useEffect(() => {
+    // Risk is accepted, meaning the user rejected the warning
```

Comment:
Took me a second to understand this part. User accepts the risk = user rejects the warning but in case it is a high severity issue the user shouldn't be able to ignore the warning anyway so if the user doesn't accept the risk we can be sure that they won't submit the transaction.